From b39e173e4b62170052aad104e91d5a7c15544f4a Mon Sep 17 00:00:00 2001
From: Cheyenne Wills <cwills@sinenomine.net>
Date: Wed, 7 May 2025 14:14:39 -0600
Subject: [PATCH 03/10] Linux: Add required MODULE_DESCRIPTION

Linux commit:
    "modpost: require a MODULE_DESCRIPTION()" (6c6c1fc09de3)
changed the Linux build process to require the MODULE_DESCRIPTION.

Add a MODULE_DESCRIPTION() to the kernel modules, including the code
template used in authconf tests.

Change-Id: Ia56cb0f778d0bc3743a1a19840c245d0af88e5ab
---
 src/afs/LINUX/osi_module.c     | 1 +
 src/afs/LINUX/osi_pag_module.c | 1 +
 src/cf/linux-test1.m4          | 1 +
 3 files changed, 3 insertions(+)

diff --git a/src/afs/LINUX/osi_module.c b/src/afs/LINUX/osi_module.c
index 0fc9beba0..599c4d83d 100644
--- a/src/afs/LINUX/osi_module.c
+++ b/src/afs/LINUX/osi_module.c
@@ -136,6 +136,7 @@ afs_cleanup(void)
 }
 
 MODULE_LICENSE("http://www.openafs.org/dl/license10.html");
+MODULE_DESCRIPTION("OpenAFS");
 module_init(afs_init);
 module_exit(afs_cleanup);
 
diff --git a/src/afs/LINUX/osi_pag_module.c b/src/afs/LINUX/osi_pag_module.c
index 7e8e4f15b..4e6c9fb94 100644
--- a/src/afs/LINUX/osi_pag_module.c
+++ b/src/afs/LINUX/osi_pag_module.c
@@ -107,6 +107,7 @@ afspag_cleanup(void)
 }
 
 MODULE_LICENSE("http://www.openafs.org/dl/license10.html");
+MODULE_DESCRIPTION("OpenAFS PAG");
 module_init(afspag_init);
 module_exit(afspag_cleanup);
 
diff --git a/src/cf/linux-test1.m4 b/src/cf/linux-test1.m4
index 010a4464f..581dc10e2 100644
--- a/src/cf/linux-test1.m4
+++ b/src/cf/linux-test1.m4
@@ -23,6 +23,7 @@ $2
 } 
 
 MODULE_LICENSE("http://www.openafs.org/dl/license10.html");
+MODULE_DESCRIPTION("openafs conftest");
 _ACEOF
     echo make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir $SPARSE_MAKEFLAGS modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD &&
     make -C $LINUX_KERNEL_BUILD M=$SRCDIR_PARENT/conftest.dir $SPARSE_MAKEFLAGS modules KBUILD_VERBOSE=1 >&AS_MESSAGE_LOG_FD 2>conftest.err &&
-- 
2.49.0

